tap_salesforce/__init__.py

```diff
@@ -98,13 +112,148 @@ def create_property_schema(field, mdata):
         mdata = metadata.write(
             mdata, ('properties', field_name), 'inclusion', 'available')
 
-    property_schema, mdata = salesforce.field_to_property_schema(field, mdata)
+    property_schema, mdata = salesforce.field_to_property_schema(
+        field, mdata, source_type)
+
+    return (property_schema, mdata)
+
+
+def create_report_property_schema(field, mdata, source_type):
+    field_name = field['label']
+
+    mdata = metadata.write(
+        mdata, ('properties', field_name), 'inclusion', 'available')
+
+    property_schema, mdata = salesforce.field_to_property_schema(
+        field, mdata, source_type)
 
     return (property_schema, mdata)
 
 
 # pylint: disable=too-many-branches,too-many-statements
 def do_discover(sf):
+    if sf.source_type == 'object':
+        do_discover_object(sf)
+    elif sf.source_type == 'report':
+        do_discover_report(sf)
+
+
+def do_discover_report(sf):
+    """Describes a Salesforce instance's reports and generates a JSON schema for each field."""
+    global_description = sf.describe_reports()
```

Comment:
As discussed, I find the double call to be confusing if we're not using the param as a filter anymore. Also makes sense to combine as just describe() and check source_type in describe() instead.